Bug in io.py when fstep 0 is not in inference results

### What happened?

A hardcoded reference to fstep 0 was introduced here: https://github.com/ecmwf/WeatherGenerator/commit/8aade866b3bc398ecc1b2372c7d77534f7dc7784#diff-580d900cb2e6a0d5c44eada1fba49b4a51df3abdf9c7771ae3ffacc6e909b1e4R402
This leads to the following:
```
  File "WeatherGenerator/packages/common/src/weathergen/common/io.py", line 402, in forecast_offset
    fstep0_datasets = self._get_datasets(self.example_key)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "WeatherGenerator/packages/common/src/weathergen/common/io.py", line 350, in _get_datasets
    group = self._get_group(key)
            ^^^^^^^^^^^^^^^^^^^^
  File "WeatherGenerator/packages/common/src/weathergen/common/io.py", line 366, in _get_group
    assert group is not None, f"Zarr group: {item.path} does not exist."
           ^^^^^^^^^^^^^^^^^
AssertionError: Zarr group: 0/ERA5strato/0 does not exist.
```

### What are the steps to reproduce the bug?

run evaluation when `forecast_offset > 0`

### Hedgedoc link to logs and more information. This ticket is public, do not attach files directly.

_No response_

--- a/packages/common/src/weathergen/common/io.py
+++ b/packages/common/src/weathergen/common/io.py
@@ -186,7 +186,7 @@ class ItemKey:
         Infer forecast offset by the (non)presence of targets at fstep 0.
 
         Args:
-            datasets: Datasets found in a fstep 0 OutputItem.
+            datasets: Datasets found in a fstep 0 OutputItem (eg. ZarrIO.example_key).
         """
         # forecast offset=1 should produce no targets at fstep 0
         return 0 if "target" in datasets else 1
@@ -364,8 +364,7 @@ class ZarrIO:
             group = self.data_root.create_group(item.path)
         else:
             try:
-                group = self.data_root.get(item.path)
-                assert group is not None, f"Zarr group: {item.path} does not exist."
+                group = self.data_root[item.path]
             except KeyError as e:
                 msg = f"Zarr group: {item.path} has not been created."
                 raise FileNotFoundError(msg) from e
@@ -406,13 +405,17 @@ class ZarrIO:
 
     @functools.cached_property
     def example_key(self) -> ItemKey:
+        fstep = 0
         try:
             sample, example_sample = next(self.data_root.groups())
             stream, example_stream = next(example_sample.groups())
-            fstep = 0
         except StopIteration as e:
             msg = f"Data store at: {self._store_path} is empty."
             raise FileNotFoundError(msg) from e
+
+        assert fstep in example_stream.groups(), (
+            "fstep 0 is missisg, but should always contain at least sources."
+        )
 
         return ItemKey(sample, fstep, stream)
 

